swf/src/types.rs

```diff
@@ -110,6 +120,16 @@ impl HeaderExt {
         }
     }
 
+    /// Returns the header for a loaded image (JPEG, GIF or PNG).
+    pub fn default_with_uncompressed_len(length: u32) -> Self {
+        Self {
+            header: Header::default_for_loaded_image(),
```

Comment:
Small nit, but I think I'd prefer to construct `Header` directly here instead of adding `Header::default_for_loaded_image`, just to avoid adding another function to the public API (`swf` is the one crate we publish onto crates.io, so it's better to avoid Ruffle-specific stuff in there when possible).

Reply:
Makes sense! I did this change and also added a test.